src/common/hook/useQueue/utils/publish.tsx

```diff
@@ -100,14 +113,20 @@ export const groupDocumentsIntoJobs = async (
   let nonce = currentNonce;
 
   const jobs: PublishingJob[] = [];
-
   // Process all verifiable documents with document store first
-  documentStoreAddresses.forEach((contractAddress) => {
+  for (const contractAddress of documentStoreAddresses) {
     const firstRawDocument = verifiableDocumentsWithDocumentStore[contractAddress][0];
     // eslint-disable-next-line @typescript-eslint/no-shadow
     const rawDocuments = verifiableDocumentsWithDocumentStore[contractAddress].map((doc) => doc.rawDocument);
-    const wrappedDocuments = wrapDocuments(rawDocuments);
+
+    const wrappedDocuments = isV3Document(rawDocuments[0])
+      ? await __unsafe__use__it__at__your__own__risks__wrapDocuments(rawDocuments)
+      : wrapDocuments(rawDocuments);
+
     const firstWrappedDocument = wrappedDocuments[0];
+    const merkleRoot = utils.isWrappedV3Document(firstWrappedDocument)
+      ? firstWrappedDocument.proof.merkleRoot
+      : firstWrappedDocument.signature?.merkleRoot;
```

Comment:
think here have already, can use this instead -> https://github.com/Open-Attestation/open-attestation/blob/master/src/shared/utils/utils.ts#L82

Reply:
done.